platform.go

```diff
@@ -73,22 +81,21 @@ type PlatformKeyDataHandler interface {
 	// data using this platform's secure device.
 	RecoverKeys(data *PlatformKeyData) (KeyPayload, error)
 
-	// RecoverKeysWithAuthValue attempts to recover the cleartext keys from the
-	// supplied data using this platform's secure device. The authValue parameter
+	// RecoverKeysWithAuthKey attempts to recover the cleartext keys from the
+	// supplied data using this platform's secure device. The key parameter
 	// is a passphrase derived key to enable passphrase support to be integrated
 	// with the secure device. The platform implementation doesn't provide the primary
 	// mechanism of protecting keys with a passphrase - this is done in the platform
 	// agnostic API. Some devices (such as TPMs) support this integration natively. For
-	// other devices, the integration should provide a way of validating the authValue in
+	// other devices, the integration should provide a way of validating the key in
 	// a way that requires the use of the secure device (eg, such as computing a HMAC of
 	// it using a hardware backed key).
-	// RecoverKeysWithAuthValue(data *PlatformKeyData, authValue []byte) (KeyPayload, error)
+	RecoverKeysWithAuthKey(data *PlatformKeyData, key []byte) (KeyPayload, error)
 
-	// ChangeAuthValue is called to notify the platform implementation that the
-	// passphrase is being changed. The oldAuthValue and newAuthValue parameters
-	// are passphrase derived keys. On success, it should return an updated
-	// PlatformKeyData.
-	// ChangeAuthValue(data *PlatformKeyData, oldAuthValue, newAuthValue []byte) (*PlatformKeyData, error)
+	// ChangeAuthKey is called to notify the platform implementation that the
+	// passphrase is being changed. The old and new parameters are passphrase derived
+	// keys. On success, it should return an updated handle.
```

Comment:
this should mention that old can be nil if this is the first time the auth key is set?

Reply:
I've updated this now